src/coreclr/src/vm/jitinterface.cpp

```diff
@@ -3481,16 +3484,24 @@ void CEEInfo::ComputeRuntimeLookupForSharedGenericToken(DictionaryEntryKind entr
 
     DictionaryEntrySignatureSource signatureSource = (IsCompilationProcess() ? FromZapImage : FromJIT);
 
+    WORD slot;
+
     // It's a method dictionary lookup
     if (pResultLookup->lookupKind.runtimeLookupKind == CORINFO_LOOKUP_METHODPARAM)
     {
         _ASSERTE(pContextMD != NULL);
         _ASSERTE(pContextMD->HasMethodInstantiation());
 
-        if (DictionaryLayout::FindToken(pContextMD->GetLoaderAllocator(), pContextMD->GetNumGenericMethodArgs(), pContextMD->GetDictionaryLayout(), pResult, &sigBuilder, 1, signatureSource))
+        if (DictionaryLayout::FindToken(pContextMD, pContextMD->GetLoaderAllocator(), 1, &sigBuilder, NULL, signatureSource, pResult, &slot))
```

Comment:
Where did this magic number `1` (and `2` below) came from? I know it was there even before your change, but I am not able to figure it out.

Reply:
For methods: we indirect once of the MethodDesc to get the m_pPerInstInfo (1)
For types: we indirect twice of the MethodTable to get the dictionary. m_pPerInstInfo is an array of dictionaries for types, due to inheritance properties.

Reply:
Ah ok ... make sense.